### ISSUE TITLE
CSS issues on certain browser/devices

### ISSUE THREAD

**Issue by baohouse** [julianguyen/ifme#561]
![android_samsung-galaxy-s4_4 3_portrait](https://user-images.githubusercontent.com/4228012/29645829-c550bffc-8834-11e7-99df-c710d1ff7f45.jpg)

Title gets clipped:
* iPhone 4S
* Opera 12 on Windows 8
* Samsung Galaxy S4

![win8_ie_10 0](https://user-images.githubusercontent.com/4228012/29645939-2c8a0868-8835-11e7-8c76-dada29053fde.png)

Background images missing:
* IE 10 & 11
* Opera 12 on Windows 8

**Comment by julianguyen** [comment]
Thanks for filing this @baohouse! Are you also interested in fixing these bugs? :P if not, can you tag it as "help wanted"? Thanks ^^

**Comment by julianguyen** [comment]
Do you notice any console errors as well?

**Comment by baohouse** [comment]
@julianguyen I ran this through Browserstack's screenshot generator, so there wasn't manual end-to-end testing to check the console.

I'll give this issue some thought. I'm attending a [Modern Web Meetup event on September 7](https://www.meetup.com/modernweb/events/241799217/), one of the topics being building a CSS pattern library, so let's see what I learn from there that I can bring back to the table. I may be good at pixel-perfect CSS, but managing CSS across multiple browsers/platforms and an ever-growing feature list for the application is still something I'm learning to get better at.

**Comment by baohouse** [comment]
Does anyone have opinions on CSS prefixes. I found https://github.com/ai/autoprefixer-rails and was wondering if this is the best solution.

**Comment by HashNotAdam** [comment]
@baohouse Compass is being used in the project and it has methods for this. In fact, I thought it does it automatically but you can change the defaults for when it stops using prefixes if you need to support browsers for longer than the defaults.

It's been a _long_ time since I looked into this so you'll want to read the docs but I believe you simply import the CSS3 module and then use the import mixins like:
```
@import "compass/css3";
...
li {
  @include display-flex(inline-flex);
}
```

What will require prefixes?

**Comment by baohouse** [comment]
@HashNotAdam Oops, I should've realized that if we use variable substitution inside CSS, then we already have a CSS enhancement library. I wanted to rule out the possibility that vendor prefixes are the cause of the CSS issues described above.

**Comment by baohouse** [comment]
Looks like the background image missing problems are solved as of #577

**Comment by baohouse** [comment]
I traced the setHeight issue back to Turbolinks, which breaks when the pageLoad() document.ready event fires. It seems to have problems dealing with the DOM Location API. In any case, https://github.com/turbolinks/turbolinks/issues/25 lists the supported browser. I'm updating the wiki   https://github.com/julianguyen/ifme/wiki/Testing with the minimum versions listed in Turbolinks, and closing the ticket for now until we have a need to support older browsers.
